### PR TITLE
Add sequence number to observed frame definition.

### LINKED ISSUE #17: Differentiate old and new values of observed address
Assume the following scenario:

* Server observes an NAT rebinding, sends "observed address" frame number 1,
* Packet carrying the observed address frame is lost
* Server observes an NAT rebinding, sends "observed address" frame number 2,
* Client receives "observed address" frame number 2,
* Server notices the loss and repeats "observed address" frame number 1,
* Client receives "observed address" frame number 1.

Thus, client is confused...

We have generally treated such issues by adding a monotonically increasing sequence number to the frame, so we can retain only the last one, or the last one for a given path. See for example the PATH_STANDBY and PATH_AVAILABLE multipath frames.

### PR DESCRIPTION
**Pull request by huitema**
Add a sequence number to the observed address frame, using the same format as previous drafts, and explain how this monotonically increasing sequence number is used to avoid acting on duplicated or delayed packets. Also strikes out the reference to handshake packets.

Close #17

### CODE REVIEW

File: draft-seemann-quic-address-discovery.md [hunk 1]
```diff
@@ -127,7 +137,7 @@ Port:
 
 : The port number, in network byte order.
 
-This frame MUST only appear in the handshake and in the application data packet
+This frame MUST only appear in the application data packet
```
Comment (marten-seemann) [inline draft-seemann-quic-address-discovery.md:140]: Let's do this in a separate PR.
```suggestion
This frame MUST only appear in the handshake and in the application data packet
```

Comment (huitema) [reply draft-seemann-quic-address-discovery.md]: OK.

**Review by marten-seemann — APPROVED** [review]
